src/test/ValidatorKeysTool_test.cpp

```diff
@@ -88,34 +87,51 @@ class ValidatorKeysTool_test : public beast::unit_test::suite
         };
 
         {
-            std::uint32_t const sequence = 1;
             std::string const expectedError = "";
-            testSeq (sequence, keyFile, expectedError);
+            testToken (keyFile, expectedError);
         }
         {
-            std::uint32_t const sequence = 1;
+            auto const keyType = KeyType::ed25519;
+            auto const kp = generateKeyPair (keyType, randomSeed ());
+
+            auto keys = ValidatorKeys (
+                keyType,
+                kp.second,
+                std::numeric_limits<std::uint32_t>::max () - 1);
+
+            keys.writeToFile (keyFile);
             std::string const expectedError =
-                boost::str (boost::format (
-                    "Sequence should exceed current sequence (%u).") %
-                    sequence);
-            testSeq (sequence, keyFile, expectedError);
-        }
-        {
-            std::uint32_t const sequence =
-                std::numeric_limits<std::uint32_t>::max ();
-            std::string const expectedError = "";
-            testSeq (sequence, keyFile, expectedError);
+                "Maximum number of tokens have already been generated.\n"
+                "Revoke validator keys if previous token has been compromised.";
+            testToken (keyFile, expectedError);
         }
         {
-            std::uint32_t const sequence =
-                std::numeric_limits<std::uint32_t>::max ();
+            createRevocation (keyFile);
             std::string const expectedError =
-                "Sequence is already at maximum value. Master keys have been revoked.";
-            testSeq (sequence, keyFile, expectedError);
+                "Validator keys have been revoked.";
+            testToken (keyFile, expectedError);
         }
     }
 
     void
+    testCreateRevocation ()
+    {
+        testcase ("Create Revocation");
+
+        using namespace boost::filesystem;
+
+        std::string const subdir = "test_key_file";
```

Comment:
Before this point you could test creating a revocation when the `validator-keys.json` file is missing.